components/promise-component.js

```diff
@@ -0,0 +1,66 @@
+// promise-component.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+module.exports = function(def){
```

Comment:
Could one of you please explain module.exports to me sometime, in the case when a function is assigned?  Web searches didn't add much light.